packages/run-protocol/src/reserve/assetReserve.js

```diff
@@ -51,24 +61,70 @@ const nonalphanumeric = /[^A-Za-z0-9]/g;
  *   marshaller: ERef<Marshaller>,
  *   storageNode: ERef<StorageNode>,
  * }} privateArgs
+ * @param {Baggage} baggage
  */
-const start = async (zcf, privateArgs) => {
+const start = async (zcf, privateArgs, baggage) => {
   /**
    * Used to look up the unique keyword for each brand, including RUN.
    *
    * @type {MapStore<Brand, Keyword>}
    */
-  const keywordForBrand = makeStore('keywords');
+  const keywordForBrand = provideDurableMapStore(baggage, 'keywordForBrand');
   /**
    * Used to look up the brands for keywords, excluding RUN because it's a special case.
    *
    * @type {MapStore<Keyword, Brand>}
    */
-  const brandForKeyword = makeStore('brands');
+  const brandForKeyword = provideDurableMapStore(baggage, 'brandForKeyword');
   /**
    * @type {MapStore<Brand, Brand>}
    */
-  const liquidityBrandForBrand = makeStore('liquidityBrands');
+  const liquidityBrandForBrand = provideDurableMapStore(
+    baggage,
+    'liquidityBrandForBrand',
+  );
+
+  const takeFeeMint = async () => {
+    if (baggage.has('runMint')) {
+      return baggage.get('runMint');
+    }
+
+    const runMintTemp = await zcf.registerFeeMint(
+      RunKW,
+      privateArgs.feeMintAccess,
+    );
+    baggage.init('runMint', runMintTemp);
+    return runMintTemp;
+  };
+  /** @type {ZCFMint} */
+  const runMint = await takeFeeMint();
+  const runKit = runMint.getIssuerRecord();
+
+  const getTotalFeeMinted = () =>
```

Comment:
this works, but all these baggage operations make me think it's worth having a singleton durable kind for the reserve contract (à la VaultDirector for VaultFactory). That's a significant refactoring so non-blocking. More of a discussion item on how we do these durability transformations.

Reply:
I moved the mutable variables into state, and left the stores outside of state.  This ended up causing me to add helper functions that will be needed by other governed contracts that use state.

Reply:
Yes, that's an interesting observation. @dtribble and I had a conversation with the same conclusion: The singleton pattern as enshrined in the singleton helpers work fine when all the per-singleton durable state is lexical `const`, i.e., never updated by assignment. But if any per-singleton state variables are updated by the singleton object, then it is better to put it into the state record, which currently requires manually writing a variation of the singleton pattern without help from the helper.